manager/integration/tests/test_snapshot.py

```diff
@@ -0,0 +1,491 @@
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    # Step 2
+    create_snapshots(client, volume, data_size)
+
+    # Step 3
+    checksums_set, generate_checksums_time = wait_for_snapshot_checksums_generate(volume) # NOQA
+    # The checksum should be calculated after creating the snapshot immediately
+    time_range = range(0, 60)
+    assert checksums_set
+    assert generate_checksums_time in time_range
+
+
+def check_hash_calculated_and_without_immediate_hash(client, volume_name, snapshot_data_integrity): # NOQA
+    """
+    1. Create and attach a volume
+    2. Create snapshots
+    3. Sleep for 120 seconds. Check snapshots' checksums are not calculated
+       and not set, because the immediate hash after snapshot creation feature
+       is disabled
+    4. Check snapshots' checksums are calculated and set by the periodic check
+       mechanism
+    """
+
+    # Step 1
+    volume = create_and_check_volume(client, volume_name, 3,
+                                     size=str(16 * Mi),
+                                     snapshot_data_integrity=snapshot_data_integrity) # NOQA
+
+    lht_hostId = get_self_host_id()
+    volume = volume.attach(hostId=lht_hostId)
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    # Step 2
+    create_snapshots(client, volume, 1)
+
+    # Step 3
+    sleep_time = 120
+    time.sleep(sleep_time)
+    assert not check_snapshot_checksums_set(volume)
+
+    # Step 4
+    checksums_set, generate_checksums_time = wait_for_snapshot_checksums_generate(volume) # NOQA
+    # +-60 is a toleration for some delays
+    remaining = SNAPSHOT_CHECK_PERIOD - sleep_time
+    time_range = range(remaining-60, remaining+60)
+    assert checksums_set
+    assert generate_checksums_time in time_range
+
+
+def check_per_volume_hash_disable(client, volume_name, snapshot_data_integrity):  # NOQA
+    """
+    1. Create and attach a volume
+    2. Create snapshots
+    3. Sleep for 300+60 seconds. Check snapshots' checksums are calculated
+       and set and also check the checksums of the snapshot disk files
+       are matched with the values from the api
+       Check snapshots' checksums are calculated by periodic snapshot
+       verification mechanism and set and also check the checksums of
+       the snapshot disk files are matched with the values from the api
+    """
+
+    # Step 1
+    volume = create_and_check_volume(client, volume_name, 3,
+                                     size=str(16 * Mi),
+                                     snapshot_data_integrity=snapshot_data_integrity)  # NOQA
+
+    lht_hostId = get_self_host_id()
+    volume = volume.attach(hostId=lht_hostId)
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    # Step 2
+    create_snapshots(client, volume, 1)
+
+    # Step 3
+    # +60 is a toleration for some delay
```

Comment:
Can you clarify why the case needs the toleration for the delay?

Reply:
Because the checksum can be calculated by `perioc check mechanism` or `immedidate calculation after the snapshot is created`. For the first case, we cannot precisely known when the event happens, so I put a toleration here and check both the time elapsed and the checksum value.

Reply:
Suggest having a `SNAPSHOT_CHECK_TOLERATION_DELAY=60` variable and add the comment to it.

Reply:
Updated. Thank you.